rclpy/rclpy/context.py

```diff
@@ -149,3 +152,18 @@ def get_domain_id(self):
             raise RuntimeError('Context must be initialized before it can have a domain id')
         with self.__context, self._lock:
             return self.__context.get_domain_id()
+
+    def __enter__(self) -> "Context":
+        # We do not accept parameters here. If one wants to customize the init() call,
+        # they would have to call it manaully and not use the ContextManager convenience
+        self.init()
+        return self
+
+    def __exit__(
+        self,
+        exc_type: type[BaseException] | None,
+        exc_val: BaseException | None,
+        exc_tb: TracebackType | None,
+    ) -> Literal[False]:
+        self.try_shutdown()
+        return False  # Reraise possible exceptions
```

Comment:
how about the following? (just a suggestion) could be more redundant?

```suggestion
        return exc_type is None  # Reraise possible exceptions
```

Reply:
Well, I think this could be explained clearer [in the Python docs](https://docs.python.org/3/library/stdtypes.html#contextmanager.__exit__). But I think always returning False is completely fine. If there was an exception, reraise it, and if not, the ouput does not matter. I like `False` because it is simple. I dpn't have strong opinions here.